awscli/customizations/cloudfront.py

```diff
@@ -10,14 +10,12 @@
 # distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
 # ANY KIND, either express or implied. See the License for the specific
 # language governing permissions and limitations under the License.
+import hashlib
```

Comment:
Thoughts on using crt's SHA1 hashing instead? https://github.com/awslabs/aws-crt-python/blob/main/awscrt/crypto.py#L10 . I'd expect it to perform better due to native bindings

Reply:
Though on the flip side I see we tend to use hashlib for similar hashes in the rest of our codebase.

Reply:
> Was any testing done to ensure the sign functions returns the same value for the same private key and message before/after this change?

Yes, I can update PR description to include an example.

> Do you think such a test would be useful?

We would still need to keep the dependency to `cryptography` in order to support such a test. However, there is no existing test that checks that an expected signing result occurs; indeed, changing to SHA256 hashing, the test still passes because all values are mocked. I can investigate adding a new test with an expected case.

> Thoughts on using crt's SHA1 hashing instead?

Sure, not opposed. Python's `hashing` is standard library, and I haven't investigated performance improvements. I wouldn't expect that much as the overhead for invoking a Python interpreter to run a CLI command is going to be much higher on a single invocation.

Reply:
It also has a more verbose interface. Usage would look like:

```
    def sign(self, message):
        hasher = Hash.sha1_new()
        hasher.update(message)

        return self.priv_key.sign(
            RSASignatureAlgorithm.PKCS1_5_SHA1,
            hasher.digest()
        )
```

Reply:
Since we use `hashlib` elsewhere I'm going to leave that change out for now.

Reply:
> We would still need to keep the dependency to cryptography in order to support such a test.

How about a comparison against a hard-coded pre-calculated hash? 

> However, there is no existing test that checks that an expected signing result occurs

My main reason for suggesting this is to ensure we did not break hash calculation by changing the underlying algorithm, and that this remains the case in the future. If that doesn't seem valuable to you I'm also okay with just a note that you manually verified this is the case.